pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -614,7 +668,18 @@ def _snstop(self, ktcond, mjrprtlvl, minimize, n, nncon, nnobj, ns, itn, nmajor,
                 callCounter = self.hist._searchCallCounter(xuser_vec)
             if callCounter is not None:
                 self.hist.write(callCounter, iterDict)
-        iabort = 0
+                # this adds funcs etc. to the iterDict by fetching it from the history file
+                iterDict = self.hist.read(callCounter)
+        snstop_handle = self.getOption("snSTOP function handle")
+        if snstop_handle is not None:
+            if not self.storeHistory:
+                raise Error("snSTOP function handle must be used with storeHistory=True")
+            iabort = snstop_handle(iterDict)
+            # if no return, assume everything went fine
+            if iabort is None:
```

Comment:
this if statement is potentially unnecessary, `iabort` is `None` if `snstop_handle` is so, and that is already been checked within this section of the code

Reply:
This is meant to catch the case where the `snstop_handle` is provided by the user, but only used for monitoring/printing and the function does not return anything. Since SNOPT demands a return code, we set it to 0 here.

Reply:
Oh, I did not realize that, nice. I think that in the longer term might be nice to have some docs/examples on `snstop_handle`, it sounds like a pretty powerful feature